### PR TITLE
Add git workflow

### PR DESCRIPTION
**Pull request by jryates**

### CODE REVIEW

**Review by krefson — COMMENTED** [review]
I strongly suggest changing 9.1 to give the command
`git fetch upstream default:default`
This not only performs the fetch but also updates the local "default" branch to sync with the upstream.  Then clauses 9.2, 9.3, 9.4 can all be removed. 

Section 12. Split into 12.a to describe using FF mode after sync OR use "squash merge"

**Comment by DBJochym** [comment]
> I strongly suggest changing 9.1 to give the command `git fetch upstream default:default` This not only performs the fetch but also updates the local "default" branch to sync with the upstream. Then clauses 9.2, 9.3, 9.4 can all be removed.
> 
> Section 12. Split into 12.a to describe using FF mode after sync OR use "squash merge"

I'm happy with these suggestions.

**Review by krefson — COMMENTED** [review]
Need one important final item in checklist:

13. Switch back to the default branch 
     ' git switch default.
     Do not do any further work on this branch and attempt any merging, or you risk creating a "sphagetti" of loops in the commit graph.  You may re-use the name of your branch if you delete the old one with "git branch -D".

File: docs/developer/git_workflow.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    1. &nbsp;
+
+          1. *Do NOT* click “Sync now” on the PR page as this does a merge and commit, which breaks our --ffonly policy
+
+          2. At your local clone, go back to step 9
+
+          3. Similar to step 10, except this is a “force” push as the rebase has changed the commit history:
+          ```
+          git push --force myfork foo
```
Comment (oerc0122) [inline docs/developer/git_workflow.md:58]: Incorrect comment
```suggestion
          git push --force myfork bar
```

**Review by krefson — APPROVED** [review]
Good enough for now!

**Pull request merged**